yasa/evaluation.py

```diff
@@ -115,123 +109,403 @@ class EpochByEpochEvaluation:
+        """
+        return self._indiv_agree_avg
+
+    @property
+    def indiv_agree_ovr(self):
+        """
+        A :py:class:`pandas.DataFrame` of ``refr_hyp``/``test_hyp`` one-vs-rest agreement scores
+        for each individual sleep session. Agreement scores are provided for each sleep stage.
+
+        .. seealso:: :py:attr:`yasa.EpochByEvaluation.indiv_agree_avg`
+        """
+        return self._indiv_agree_ovr
+
+    @staticmethod
+    def multi_scorer(df):
+        """Compute multiple agreement scores from a 2-column dataframe.
+
+        This function offers convenience when calculating multiple agreement scores using
+        :py:meth:`pandas.DataFrame.groupby.apply`. Scikit-learn doesn't include a function that
+        return multiple scores, and the GroupBy implementation of ``apply`` in pandas does not
+        accept multiple functions.
+
+        Parameters
+        ----------
+        df : :py:class:`pandas.DataFrame`
+            A :py:class:`pandas.DataFrame` with exactly 2 columns and length of *n_samples*.
+            The first column contains true values and second column contains predicted values.
 
         Returns
         -------
-        agreement : :py:class:`pandas.Series`
-            A :py:class:`pandas.Series` with agreement metrics as indices.
+        scores : dict
+            A dictionary with scorer names (``str``) as keys and scores (``float``) as values.
         """
-        true = self.hypno_ref.hypno.to_numpy()
-        pred = self.hypno_test.hypno.to_numpy()
-        accuracy = metrics.accuracy_score(true, pred)
-        kappa = metrics.cohen_kappa_score(true, pred)
-        jaccard = metrics.jaccard_score(true, pred, average="weighted")
-        precision = metrics.precision_score(true, pred, average="weighted", zero_division=0)
-        recall = metrics.recall_score(true, pred, average="weighted", zero_division=0)
-        f1 = metrics.f1_score(true, pred, average="weighted", zero_division=0)
-        scores = {
-            "accuracy": accuracy,
-            "kappa": kappa,
-            "weighted_jaccard": jaccard,
-            "weighted_precision": precision,
-            "weighted_recall": recall,
-            "weighted_f1": f1,
+        t, p = zip(*df.values)  # Same as (df["col1"], df["col2"]) but teensy bit faster
+        ## Q: The dictionary below be compiled more concisely if we were comfortable accessing
+        ##    "private" attributes. I understand that's a no-no but I'm not exactly sure why.
+        ##     For example:
+        ##     >>> scorers = ["accuracy", "recall"]
+        ##     >>> funcs = { s: skm.__getattribute__(f"{s}_scorer") for s in scorers }
+        ##     >>> scores = { s: f(true, pred) for s, f in funcs.items() }
+        ##     Keywords could be applied as needed by checking f.__kwdefaults__
+        ##     This would offer an easy way for users to add their own scorers with an arg as well.
```

Comment:
Totally fine with me

Reply:
I might switch to this. There is already updated code in this section now, but I think I'd like to consolidate in this way. Leaving this hanging for now as a reminder.